### PR TITLE
OGC API - Processes CITE compliance

### PR DESCRIPTION
**Pull request by tomkralidis**
# Overview
This PR covers pygeoapi compliance against OGC API - Processes - Part 1: Core.  The PR is marked as WIP until all [CITE](https://github.com/opengeospatial/ets-ogcapi-processes10) tests pass.
# Related Issue / Discussion
None
# Additional Information
None
# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by ricardogsilva** [comment]
@tomkralidis 

Are you committing to adding the missing bits of OAProc in order to align with v1.0.0 of the standard with this PR?

Or will you be accepting PRs for adding missing stuff and then rebasing this PR regularly and checking whether CITE eventually passes?

I just want to know whether I should focus on adding some of the missing OAProc features, or if you'd want to take that task on yourself.

**Comment by ricardogsilva** [comment]
also, related:

- #1285 
- #1316 
- #1317

**Comment by tomkralidis** [comment]
FYI PR rebased and updated.  Fixing remaining issues (down to 1 failure against ETS).

File: tests/cite/cite.config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ server:
     bind:
         host: 0.0.0.0
         port: 5001
-    url: http://localhost:5001
+    url: http://172.19.131.164:5001
```
Comment (kalxas) [inline tests/cite/cite.config.yml:5]: Is this IP supposed to be here?

**Review by kalxas — APPROVED** [review]

File: tests/cite/cite.config.yml (hunk 1 above)
Comment (tomkralidis) [reply tests/cite/cite.config.yml]: Good catch! Fixed.

Comment (kalxas) [reply tests/cite/cite.config.yml]: Thanks

**Review by kalxas — APPROVED** [review]

**Pull request merged**